library/ecp.c

```diff
@@ -3288,7 +3288,10 @@ int mbedtls_ecp_read_key(mbedtls_ecp_group_id grp_id, mbedtls_ecp_keypair *key,
         MBEDTLS_MPI_CHK(mbedtls_mpi_read_binary(&key->d, buf, buflen));
     }
 #endif
+#if defined(MBEDTLS_ECP_MONTGOMERY_ENABLED) || \
+    defined(MBEDTLS_ECP_SHORT_WEIERSTRASS_ENABLED)
     MBEDTLS_MPI_CHK(mbedtls_ecp_check_privkey(&key->grp, &key->d));
+#endif
```

Comment:
Functionally, this doesn't change anything. The only supported curves are Montgomery curves and short Weierstrass curves. If the preprocessor condition is false, `mbedtls_ecp_group_load` above can't have succeeded.

If we add a new curve family (e.g. twisted Edwards), we'll have to add a case for it in this function and many others.

If we somehow add support for a new curve family to `mbedtls_ecp_group_load` but not here, then this function won't set `key→d`. Note that this is not an uninitialized variable: this function assumes that `key` is already initialized with `mbedtls_ecp_keypair_init`. If `key` is freshly initialized, then `key→d` has the value 0 (represented with length 0), and `mbedtls_ecp_check_privkey` will either succeed or fail depending on whether `{0, {}}` is considered a valid private key object. If `key` had been in use before, then `key→d` still has its old value and may or may not pass `mbedtls_ecp_check_privkey`.

I don't see any advantage with this change. In the unlikely chance that we forget this function when we add support for a new curve family, we're replacing a call that has a good chance of returning an error with nothing.